### PR TITLE
Specify start index of file name in GenerateDataCommand

### PR DESCRIPTION
**Pull request by vrajat**
`GenerateDataCommand` can generate data in multiple files. It is useful to use the filename to generate unique segment names. For example, the in the following ingestion job spec, the integer suffix is used to generate the segment name.

```
segmentNameGeneratorSpec:
  type: inputFile
  configs:
    file.path.pattern: '.+_(.+)\.csv'
    segment.name.template: 'nation_OFFLINE_\${filePathPattern:\1}'
```

Another useful requirement is to generate data in batches. For example, generate data in batches of 10000 files. By default, the suffix of the file path starts from 0. Therefore data file names and consequently the segment names are not unique. Unique file names can be generated by specifying start index using the `-fileIndex` argument. Therefore, the following sequence of commands will allow a user to generate and ingest data in batches.

```
pinot-admin.sh  GenerateData ... -fileIndex 0
# Launch ingestion job
pinot-admin.sh  GenerateData ... -fileIndex 10000
#Launch ingestion job
....
```

This is another option to generate segments with unique names along with other techniques such as UUID and timestamp.

### CODE REVIEW

File: pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/GenerateDataCommand.java [hunk 1]
```diff
@@ -120,9 +123,9 @@ public boolean execute()
     gen.init(spec);
 
     if (FORMAT_AVRO.equalsIgnoreCase(_format)) {
-      DataGenerationHelpers.generateAvro(gen, _numRecords, _numFiles, _outDir, _overwrite);
+      DataGenerationHelpers.generateAvro(gen, _numRecords, _numFiles, _outDir, _overwrite, _fileIndex);
     } else if (FORMAT_CSV.equalsIgnoreCase(_format)) {
-      DataGenerationHelpers.generateCsv(gen, _numRecords, _numFiles, _outDir, _overwrite);
+      DataGenerationHelpers.generateCsv(gen, _numRecords, _numFiles, _outDir, _overwrite, _fileIndex);
```
Comment (yashmayya) [inline pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/GenerateDataCommand.java:128]: Why is this not applicable to the JSON generator?

**Review by yashmayya — COMMENTED** [review]
> With this feature, it is possible to generate a dataset across multiple runs of GenerateDataCommand. The starting number can be specified using -fileIndex argument. Default is 0.

The changes look fine to me but what's the use case for this exactly?

File: pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/GenerateDataCommand.java (hunk 1 above)
Comment (vrajat) [reply pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/GenerateDataCommand.java:128]: Missed a spot! Added

**Comment by vrajat** [comment]
> The changes look fine to me but what's the use case for this exactly?

I've updated the description. Tell me if it is more informative.

**Review by yashmayya — APPROVED** [review]

**Pull request merged**